interview-questions.md

```diff
@@ -9,59 +9,58 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 
 
 4. What is JSX? What is one syntax difference between HTML and JSX?
 
-  Your answer:
+  Your answer: JSX is used with React and responsible for those familiar with HTML syntax to do similiar work through the Javascript environment. With JSX, one has to import or point the child components back to the parent edition on the original App.js file which stores the bulk of the information. 
 
-  Researched answer:
+  Researched answer: JSX is JavaScript + XML or a React version of HTML, allows the use of HTML-like syntax that can be used to create React components. Without JSX, it would more difficult to create largely nested HTML documents using JS only syntax. Speaking of syntax, a major difference between native HTML and JSX is with JSX the word "class" cannot be used to identify HTML/CSS classes, since that word is reserved in JavaScript, so as an alternative or work around, instead one would use - "className". 
 
 
 
 5. What is yarn? What file(s) are modified in a React application when you run yarn?
 
-  Your answer:
+  Your answer: Yarn is a JS file manager that is used to connect your JS files to a node so that one's changes can be viewed in a browser. When you run yarn your App.js and node files are modified becauses they are directly effected by the changes or edits done when writing code through React/JS. 
 
-  Researched answer:
+  Researched answer: Yarn is package file manager, originally released from Facebook that is used with React and Jest. When running yarn through a React application the "package.json" and "yarn.lock" files are modified because are both important pieces for connecting the node and active files. 
 
 
 
 6. STRETCH: What is an anonymous function in JavaScript?
 
-  Your answer:
+  Your answer: 
 
   Researched answer:
 
 
 ## Looking Ahead: Terms for Next Week
 
-1. Conditional rendering:
+1. Conditional rendering: applying conditions in the same manner you would with regular Javascript like using conditional operators or JS operators like `if` to create live element representation that thru React can be udated to mirror the user interface.
 
-2. Object-oriented programming:
+2. Object-oriented programming: centered around object creation that contains data & methods, this helps lessen repetition of code and thus cleaner.
 
-3. Ruby:
+3. Ruby: open source pprogramming language thats designed to be more humancenteric, meaning the syntax is mire natural to read and easy to write having a focus around simpliicity yet pproductivity. 
 
-4. Ruby blocks:
+4. Ruby blocks: allows programmers more freedom and flexible when it comes to code because the functions are annoymous in nature and can be passed into methods.
 
-5. Ruby hashes:
+5. Ruby hashes: aka associative arrays is a collection of individualized keys and their relative values, the layout is like an array but instead of having integers, programmers have the ability to use any object type as its index.
```

Comment:
Fantastic answers again, Steven!  I love your use of examples - it really shows your understanding.  Keep it up!